src/components/editor/InputListLOC.jsx

```diff
@@ -50,13 +50,22 @@ class InputListLOC extends Component {
   }
 
   render() {
-    let lookupUri, isMandatory, isRepeatable
+    let isMandatory, isRepeatable
+
+    if (this.state.lookupUri === undefined) {
+      console.error("lookup URI was not defined!")
+    }
+
     try {
       isMandatory = JSON.parse(this.props.propertyTemplate.mandatory)
+    } catch (error) {
+      //We do not really care about this error. If there is an error the variable will be undefined and hence false
```

Comment:
What causes the errors on lines 62 and 68? I wonder if we want to use a different strategy here, lest we eat errors that we don't anticipate. I'm a bit leery of using `try`/`catch` for flow control. For instance, if the situation we're coding around here is a property template that hasn't set the mandatory attribute, perhaps we should replace this with:

```javascript
const isMandatory = this.props.propertyTemplate.mandatory !== undefined ? this.props.propertyTemplate.mandatory : false
```

Reply:
So theoretically what if they set the mandatory: "foo" and/or repeatable: "bar"? They are not undefined. The Typeahead component does not break but throws a warning: 

 ```
Warning: Failed prop type: Invalid prop `multiple` of type `string` supplied to `TypeaheadContainer(WrappedTypeahead)`, expected `boolean`.
```

The schema validation probably checks that those values are "true" or "false", but we also have clever librarians who like to hand-edit their resource templates.

I like the syntax you provided and I'm happy to make that change but I'd like to get your thoughts first 👆.

Reply:
@jgreben for added robustness, do you think we ought to add a new function to the `Utilities` class? something like:

```javascript
export const booleanPropertyFromTemplate = (template, key, defaultvalue) => {
  // Use safe navigation for dynamic properties: https://github.com/tc39/proposal-optional-chaining#syntax
  const propertyValue = template?.[key]

  if (!propertyValue)
    return defaultValue

  parsedValue = JSON.parse(propertyValue)
  
  if (parsedValue !== true && parsedValue !== false)
    return defaultValue

  return parsedValue
}
```

:question:

Reply:
I like this, but how it will work with (e.g.) `this.props.propertyTemplate.valueConstraint.repeatable`?

Reply:
@jgreben In that context, I reckon it could be invoked thusly:

```javascript
booleanPropertyFromTemplate(this.props.propertyTemplate.valueConstraint, 'repeatable', false)
```

So maybe the function and first arg could be renamed so it's clear that a property template is not required, but rather any ol' object?

Reply:
Ah nevermind, I was thinking that `template` was always the whole property template, but in the second case it would actually be the valueConstraint object. Maybe template is not the right var name. I will add this to the next commit and you can judge.

Reply:
awesome! thanks, @jgreben

Reply:
I don't like that the optional chaining syntax makes my Webstorm editor complain...but I guess that's my own problem!

Reply:
Emacs complains at me too. Ha!